### ISSUE TITLE
`app_id` In documentation is ambiguous for MongoDB Atlas Event Trigger

### ISSUE THREAD

**Issue by TylerSustare** [mongodb/terraform-provider-mongodbatlas#957]
## tl;dr;
The `app_id` to create an _Atlas Event Trigger_ [in the documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/event_triggers#app_id) is ambiguous. What is **your application** in the doc talking about here?
```
app_id - (Required) The ObjectID of your application.
```

### Terraform CLI and Terraform MongoDB Atlas Provider Version
```
$ terraform --version
Terraform v1.3.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.45.0
+ provider registry.terraform.io/mongodb/mongodbatlas v1.6.1
```

### Terraform Configuration File 

```hcl
# database to keep track of all otters on earth

provider "mongodbatlas" {
  public_key  = var.atlas_public_key
  private_key = var.atlas_private_key
}
resource "mongodbatlas_project" "otter_atlas" {
  name   = "otter-otter"
  org_id = var.atlas_org_id
}
resource "mongodbatlas_advanced_cluster" "otter_free" {
  project_id   = mongodbatlas_project.otter_atlas.id
  name         = "otter-otter"
  cluster_type = "REPLICASET"

  replication_specs {
    region_configs {
      electable_specs {
        instance_size = "M0" # free
      }
      provider_name         = "TENANT"
      backing_provider_name = "AWS"
      region_name           = var.atlas_region
      priority              = 1
    }
  }
  depends_on = [mongodbatlas_project.otter_atlas]
}

resource "mongodbatlas_database_user" "db-user" {
  username           = var.atlas_dbuser
  password           = var.atlas_dbpassword
  auth_database_name = "admin"
  project_id         = mongodbatlas_project.otter_atlas.id
  roles {
    role_name     = "readWriteAnyDatabase"
    database_name = "admin"
  }
  depends_on = [mongodbatlas_project.otter_atlas]
}

resource "mongodbatlas_event_trigger" "otter" {
  project_id             = mongodbatlas_project.otter_atlas.id
  app_id                 = mongodbatlas_advanced_cluster.otter_free.id
  name                   = "Event trigger to AWS Event Bridge"
  type                   = "DATABASE"
  disabled               = false
  config_operation_types = ["INSERT", "UPDATE", "REPLACE", "DELETE"]
  config_database        = "otterotter"
  config_collection      = "otter"
  config_service_id      = "1"
  config_full_document   = true
  event_processors {
    aws_eventbridge {
      config_region     = var.aws_region
      config_account_id = var.aws_account_id
    }
  }
  depends_on = [mongodbatlas_project.otter_atlas, mongodbatlas_advanced_cluster.otter_free]
}

data "mongodbatlas_event_triggers" "otter" {
  project_id = mongodbatlas_event_trigger.otter.project_id
  app_id     = mongodbatlas_event_trigger.otter.app_id
}

```

### Steps to Reproduce
1. `$ terraform apply -auto-approve`

### Expected Behavior
1. Created the database trigger with the cluster's ID.
2. Helpful documentation on what the `app` should be relating to the `app_id`

### Actual Behavior
```
│ Error: error creating MongoDB EventTriggers (63935a667e4e335fe1d47e63): POST https://realm.mongodb.com/api/admin/v3.0/groups/63935a667e4e335fe1d47e63/apps/Y2x1c3Rlcl9pZA==:NjM5MzVhNmIwNWUxMTI3M2IzZmFhOGE4-Y2x1c3Rlcl9uYW1l:ZmVlZGJhY2stb3R0ZXI=-cHJvamVjdF9pZA==:NjM5MzVhNjY3ZTRlMzM1ZmUxZDQ3ZTYz/triggers: 404 (request "") app not found
│ 
│   with mongodbatlas_event_trigger.otter,
│   on atlas.tf line 60, in resource "mongodbatlas_event_trigger" "otter":
│   60: resource "mongodbatlas_event_trigger" "otter" {
```
### Additional Context

Not needed.

### References
According to [the doc for event triggers](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/event_triggers) we need the `app_id` but the example shows a string as an example. In the `Argument Reference` section it doesn't help to know what the Application should be.

**Comment by TylerSustare** [comment]
Based on the `POST` attempt it looks like I need to create a `realm` app. But based on creating an event trigger in the console, that doesn't seem like it would be needed to create a trigger in terraform.

**Comment by Zuhairahmed** [comment]
Thanks @TylerSustare. To learn more about what is and how to pull `app_id` see here: https://www.mongodb.com/docs/atlas/app-services/admin/api/v3/#section/Project-and-Application-IDs 
Also you are right, we can make this more clear in our docs. Will add this supporting link to help future users. Appreciate the feedback here.

**Comment by Zuhairahmed** [comment]
Doc update in internal tracking [INTMDB-517](https://jira.mongodb.org/browse/INTMDB-517). Should go out as part of near term release.

**Comment by TylerSustare** [comment]
Thank you @Zuhairahmed 😄 I appreciate the link to how to get the `app_id` via the Atlas App Services API. And that was incredibly quick!

My real question, and what I would hope to get out of the docs though is: **what mongodb resource would have this `app_id` in order to not have to go outside of terraform to use the Atlas App Services API to get the `app_id`?**

**Comment by TylerSustare** [comment]
For example, I would assume that based on the "Atlas App Services API" getting the app ID from the group/cluster ID then I could use the `cluster_id` from `mongodbatlas_advanced_cluster`

When I try to update the TF file like this I still get the same error, just with the cluster ID.

```hcl
resource "mongodbatlas_event_trigger" "otter" {
  project_id             = mongodbatlas_project.otter_atlas.id
  app_id                 = mongodbatlas_advanced_cluster.otter_free.cluster_id 
```

**Comment by Zuhairahmed** [comment]
@TylerSustare thanks for highlighting error, this is going to require a code change. Will include as part of [INTMDB-517](https://jira.mongodb.org/browse/INTMDB-517) and let you know when issue has been resolved. Feel free to reach out if you need anything else in the interim.

**Comment by TylerSustare** [comment]
bump

**Comment by Ballymultani** [comment]
Hi, is there any update on this please? We're facing a similar problem.

**Comment by Zuhairahmed** [comment]
hi @Ballymultani @TylerSustare we should have an update on this soon. feel free to reach out if you need anything else in the interim.

**Comment by Ballymultani** [comment]
Hi, any update?

**Comment by Futhark** [comment]
Is there any plan to add App Services resource types?

**Comment by Zuhairahmed** [comment]
@TylerSustare @Ballymultani fix for this has been picked up for v1.8.1 sprint which should be released in ~3 weeks. @Futhark yes we are planning to integrate Atlas App Services into Atlas Admin API/Terraform Provider, but this is a large effort and still several quarters away. Feel free to reach out in future and can share updates if you need.

**Comment by Zuhairahmed** [comment]
@TylerSustare @Ballymultani @Futhark  v1.8.1 has been published to the [Terraform Public Registry](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest)! take a spin and let us know if you need anything else.

**Comment by Zuhairahmed** [comment]
closing this issue, but feel free to re-open if you need anything else here

**Comment by Alexkuva** [comment]
Hi,

Since the 1.8.1 release how to get app_id (without leaving terraform or adding static values) please?
Neither mongodbatlas_advanced_cluster.otter_free.cluster_id  or mongodbatlas_advanced_cluster.otter_free.id works

**Comment by adamMo** [comment]
@Alexkuva did you get the answer to your question? 
Currently I am struggling with the same issue and the only thing I see is keeping this app_id in tfvars but it sounds ugly and defies purpose of terraform...

**Comment by Alexkuva** [comment]
I discovered a workaround using a module that automates the entire process with HTTP data. I’ve opened an issue to detail my approach: [here](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1949) where I explain how I implemented the workaround.

**Comment by adamMo** [comment]
Thanks a lot @Alexkuva, works like a charm!

**Comment by remiflament** [comment]
Thanks for @Alexkuva , it's still useful in 2025

**Comment by SebasAnasco1517** [comment]
Thanks @Alexkuva, it saved me a lot of time!!!
